vcd/datasource_vcd_nsxt_edgegateway.go

```diff
@@ -105,27 +115,110 @@ func datasourceVcdNsxtEdgeGatewayRead(ctx context.Context, d *schema.ResourceDat
 	log.Printf("[TRACE] NSX-T edge gateway datasource read initiated")
 
 	vcdClient := meta.(*VCDClient)
-
-	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	org, err := vcdClient.GetOrgFromResource(d)
 	if err != nil {
 		return diag.FromErr(fmt.Errorf("error retrieving Org: %s", err))
 	}
 
-	if vdc.IsNsxv() {
-		return diag.Errorf("please use 'vcd_edgegateway' for NSX-V backed VDC")
+	// Validate if VDC or VDC Group is NSX-T backed
+	inheritedVdcField := vcdClient.Vdc
+	vdcField := d.Get("vdc").(string)
+	ownerIdField := d.Get("owner_id").(string)
+	usedFieldId, err := pickVdcIdByPriority(org, inheritedVdcField, vdcField, ownerIdField)
+
+	if err != nil {
+		return diag.Errorf("error finding VDC ID: %s", err)
 	}
 
-	edge, err := vdc.GetNsxtEdgeGatewayByName(d.Get("name").(string))
+	isNsxt, err := isBackedByNsxt(org, usedFieldId)
 	if err != nil {
-		return diag.FromErr(fmt.Errorf("could not retrieve NSX-T edge gateway: %s", err))
+		return diag.Errorf("error ")
+	}
+
+	if !isNsxt {
+		return diag.Errorf("please use 'vcd_edgegateway' for NSX-V backed VDC")
+	}
+	// EOF validate if VDC or VDC Group is NSX-T backed
```

Comment:
This comment and its placement is strange. "EOF"?

Reply:
I wanted to explicitly mark the code (start and end of it) which just checks that we are dealing with NSX-T (not NSX-V) Edge Gateway. This was a very simple check before, but as we got into VDC Groups it got a lot more complicated mainly because of backwards compatibility.

I have now moved this check to another function.

Reply:
Ah, I have not realized it's the bottom of the block (IIRC, the upper part was not visible in GitHub by default).